runtime/lua/vim/lsp/protocol.lua

```diff
@@ -957,6 +961,23 @@ function protocol.resolve_capabilities(server_capabilities)
     error("The server sent invalid implementationProvider")
   end
 
+  local workspace_properties
+  local workspaceFolderCapabilities = server_capabilities.WorkspaceFoldersServerCapabilities
```

Comment:
I'm not actually sure if language servers return this by default... these should be false, but otherwise we're going to have to manually flag the capability in each language server